// @aws-sdk/client-ivs/dist/es/commands/GetChannelCommand.js
import { __extends } from "tslib";
import { GetChannelRequest, GetChannelResponse } from "../models/models_0";
import { deserializeAws_restJson1GetChannelCommand, serializeAws_restJson1GetChannelCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Gets the channel configuration for the specified channel ARN. See also <a>BatchGetChannel</a>.</p>
 */
var GetChannelCommand = /** @class */ (function (_super) {
    __extends(GetChannelCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function GetChannelCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    GetChannelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IvsClient";
        var commandName = "GetChannelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetChannelRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetChannelResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetChannelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetChannelCommand(input, context);
    };
    GetChannelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetChannelCommand(output, context);
    };
    return GetChannelCommand;
}($Command));
export { GetChannelCommand };
